DittoHealth/src/main/java/live/ditto/health/usecase/GetDittoMissingPermissionsFlow.kt

```diff
@@ -0,0 +1,21 @@
+package live.ditto.health.usecase
+
+import android.content.Context
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.isActive
+import live.ditto.transports.DittoSyncPermissions
+import kotlin.coroutines.coroutineContext
+
+class GetDittoMissingPermissionsFlow(
+    private val context: Context,
+) {
+    operator fun invoke(
+        timeoutMs: Long = 300
+    ) = flow {
+        while (coroutineContext.isActive) {
```

Comment:
Looks good, wonder if in future we can do this without looping? Perhaps a permissions listener or Android Broadcast Receiver of some sort

Reply:
Permissions are not observables in Android.
I am afraid is not other way to do it :(